storm-core/src/jvm/org/apache/storm/topology/TopologyBuilder.java

```diff
@@ -179,8 +188,13 @@ public BoltDeclarer setBolt(String id, IRichBolt bolt) throws IllegalArgumentExc
     public BoltDeclarer setBolt(String id, IRichBolt bolt, Number parallelism_hint) throws IllegalArgumentException {
         validateUnusedId(id);
         initCommon(id, bolt, parallelism_hint);
-        _bolts.put(id, bolt);
-        return new BoltGetter(id);
+
+        if (_bolts.containsKey(id)) {
+            throw new IllegalArgumentException("Bolt " + id + " have existed");
```

Comment:
Can we say `"duplicate bolt id found " + id` instead?  I think it sounds a bit cleaner.